src/TypeSpec.Extension/Emitter.Csharp/src/lib/model.ts

```diff
@@ -667,40 +666,62 @@ export function getInputType(
             logger.info(
                 `No specified type for discriminator property '${model.DiscriminatorPropertyName}'. Assume it is a string.`
             );
-            const discriminatorProperty = {
+            const discriminatorProperty: InputModelProperty = {
                 Name: model.DiscriminatorPropertyName,
                 SerializedName: model.DiscriminatorPropertyName,
                 Description: "Discriminator",
                 IsRequired: true,
                 IsReadOnly: false,
-                IsNullable: false,
                 Type: {
                     Name: "string",
                     Kind: InputTypeKind.String,
                     IsNullable: false
                 } as InputPrimitiveType,
                 IsDiscriminator: true
-            } as InputModelProperty;
+            };
             // put default discriminator property as the first property to keep previous behavior
             outputProperties.unshift(discriminatorProperty);
         }
     }
-    function getInputModelBaseType(m?: Model): InputModelType | undefined {
-        if (!m) {
-            return undefined;
-        }
 
-        // Arrays and dictionaries can't be a base type
-        if (m.indexer) {
-            return undefined;
+    function getInputModelBaseType(
+        m: Model
+    ): [InputModelType | undefined, InputDictionaryType | undefined] {
```

Comment:
Can we return the union of  'InputModelType |  InputDictionaryType | undefined' for this method?

Reply:
Then we will have to write this in the place where we call this:
```
if (baseModel === undefined)
{
// undefined case
}
else if (isModelType(baseModel))
{
// do base model
}
else if (isInputDictionaryType(baseModel))
{
// do inherited dictionary type
}
```
I believe this looks way more ugly.